controllers/datadogagent/override/podtemplatespec.go

```diff
@@ -6,22 +6,169 @@
 package override
 
 import (
-	corev1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/util/errors"
+	"fmt"
+	"strings"
 
+	apicommon "github.com/DataDog/datadog-operator/apis/datadoghq/common"
 	"github.com/DataDog/datadog-operator/apis/datadoghq/v2alpha1"
 	"github.com/DataDog/datadog-operator/controllers/datadogagent/feature"
+	"github.com/DataDog/datadog-operator/controllers/datadogagent/object/volume"
+	corev1 "k8s.io/api/core/v1"
 )
 
 // PodTemplateSpec use to override a corev1.PodTemplateSpec with a 2alpha1.DatadogAgentPodTemplateOverride.
-func PodTemplateSpec(manager feature.PodTemplateManagers, override *v2alpha1.DatadogAgentComponentOverride) (*corev1.PodTemplateSpec, error) {
-	// TODO(operator-ga): implement OverridePodTemplate
+func PodTemplateSpec(manager feature.PodTemplateManagers, override *v2alpha1.DatadogAgentComponentOverride, componentName v2alpha1.ComponentName, ddaName string) {
+	// Notice that there are several attributes in a
+	// v2alpha1.DatadogAgentComponentOverride, like "Replicas" or "Disabled",
+	// that are not related with a pod template spec. The overrides for those
+	// attributes are not applied in this function.
+
+	if override == nil {
+		return
+	}
+
+	// TODO: seccomprootpath, seccompcustomprofile, seccomprofilename.
+	// They should only apply to system-probe, and I think there's some setup
```

Comment:
fyi reviewers

Reply:
should it be part of a feature "deployment security", and I think it is at container lever that it need to be done.

which change we need to do at Pod level?